src/Spice86.Shared/Emulator/VM/Breakpoint/Serializable/SerializedBreakpoint.cs

```diff
@@ -0,0 +1,50 @@
+﻿namespace Spice86.Shared.Emulator.VM.Breakpoint.Serializable;
+
+using System.Text.Json.Serialization;
+
+/// <summary>
+/// Base class for serializable breakpoint data.
+/// </summary>
+[JsonPolymorphic(TypeDiscriminatorPropertyName = "$type")]
+[JsonDerivedType(typeof(SerializedBreakpoint), typeDiscriminator: "breakpoint")]
+[JsonDerivedType(typeof(SerializedBreakpointRange), typeDiscriminator: "range")]
+public record SerializedBreakpoint {
```

Comment:
The record name is a bit misleading. I would call it SerializableBreakpoint

Reply:
Done.